commands/create/kubeconfig/command.go

```diff
@@ -443,7 +444,9 @@ func createKubeconfig(ctx context.Context, args Arguments) (createKubeconfigResu
 	} else {
 		// create a self-contained kubeconfig
 		var yamlBytes []byte
-		logger, err := micrologger.New(micrologger.Config{})
+		logger, err := micrologger.New(micrologger.Config{
+			IOWriter: new(bytes.Buffer), // to suppress any log output
+		})
 		if err != nil {
```

Comment:
Is there a better way to achieve this?

Reply:
Not sure if there is a better way to do this, I would be interested why this suddenly became a problem.

Reply:
Don't think it happened suddenly. It might just have been unnoticed. This part of the code is only executed when the `--self-contained` flag is used. Also, `k8srestconfig` hasn't always been used for this.

Reply:
Yes `k8srestconfig` was introduced here. https://github.com/giantswarm/gsctl/pull/351

We created the `giantswarm/kubeconfig` library to generate kubeconfigs for app CRs. So the idea was to reuse it in gsctl instead of maintaining 2 copies of the structs etc.

Maybe we can disable the logging in `k8srestconfig` as an alternative to suppressing the output?

Reply:
Having a setting to surpress info logs would be nice. Similar to the loglevel we have in e.g. opsctl / devctl.

Reply:
Now that this is discussed, can this get approval, please?